src/signature.js

```diff
@@ -26,20 +26,33 @@ HMP.parse('hello :: Foo a => a -> String');
 
 // type TypeMap = StrMap Type
 
-//-----------------------------------------------------------------------------
+// ----------------------------------------------------------------------------
 //
 // Utilities
 //
-//-----------------------------------------------------------------------------
-
-const lift = R.map;
-const lift2 = R.liftN(2);
-const uncurry2 = R.uncurryN(2);
-
-// :: String -> String
-const stripNamespace = R.compose(R.last, R.split('/'));
-
-// :: Number -> String
+// ----------------------------------------------------------------------------
+
+// isEmpty :: Foldable f => f a -> Boolean
+const isEmpty = xs => xs.length === 0;
+// propEq :: String -> a -> StrMap a -> Boolean
+const propEq = prop => val => obj => obj[prop] === val;
+// indexBy :: (StrMap a -> String) -> Array (StrMap a) -> StrMap (StrMap a)
+const indexBy = memoize
+  (f => S.reduce
+    (xs => x => S.insert
+      (f (x))
+      (x)
+      (xs))
+    ({}));
+
+// stripNamespace :: String -> String
+const stripNamespace = memoize (S.pipe ([
+  S.splitOnRegex (/\//g),
+  S.last,
+  unchecked.fromMaybe (undefined),
```

Comment:
Don't use `S.last` here, as we know the array will never be empty. Use `ss => ss[ss.length - 1]` instead.

Reply:
Fair point. Will do the swap. This is another one of those `R.compose` utility functions I replaced with a `sanctuary`-implemented one, attempting to keep the exact behavior of the original function. At the time, I didn't think so far as whether `S.splitOnRegex` (or `S.splitOn`) could ever return an empty array, and I just went for the same option.